pkg/db/glogrus/glogrus.go

```diff
@@ -0,0 +1,107 @@
+// Package glogrus provides a gorm logger that wraps a logrus.Logger.
+package glogrus
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+	glogger "gorm.io/gorm/logger"
+	gutils "gorm.io/gorm/utils"
+)
+
+// Config is used to configure a gorm Logger that wraps a logrus.Logger.
+type Config struct {
+	// Logger is the logrus logger to use. If nil, logrus.StandardLogger() is used.
+	Logger *logrus.Logger
+
+	// SlowThreshold is the threshold for logging slow queries. If zero, 500ms is used.
+	SlowThreshold time.Duration
+
+	// IgnoreRecordNotFoundError determines if `gorm.ErrRecordNotFound` errors are logged.
+	// `gorm.ErrRecordNotFound` logging is disabled IFF IgnoreRecordNotFoundError is true.
+	IgnoreRecordNotFoundError bool
+}
+
+// New returns a new *Logger configured with the given config.
+func New(cfg Config) *Logger {
+	return &Logger{
+		logger:                    cfg.Logger,
+		slowThreshold:             cfg.SlowThreshold,
+		ignoreRecordNotFoundError: cfg.IgnoreRecordNotFoundError,
+	}
+}
```

Comment:
What are your thought about doing something like this?
```suggestion
func New(cfg Config) logger.Interface {
	l := &logger{
		logger:                    cfg.Logger,
		slowThreshold:             cfg.SlowThreshold,
		ignoreRecordNotFoundError: cfg.IgnoreRecordNotFoundError,
	}
	l.complete()
	return l
}
```

Reply:
The only reason I didn't was because `l.complete()` will get called lazily later anyway.

But now that you point it out, this will give us a tiny optimization for Loggers initialized via `New`. Sounds reasonable to me.

Reply:
Sorry, I didn't notice that you also made the type private and made the return type an interface. Let me address that now.

As a rule-of-thumb for Go, I think it's generally better to return structs instead of interfaces. Here's [a good summary of the reasoning](https://bryanftan.medium.com/accept-interfaces-return-structs-in-go-d4cab29a301b), and although we're not defining a net new interface, I think the pattern still provides more flexibility.

Reply:
To be honest, I was trying to get rid of the need to call `l.complete()` in every method. However, I think this is fine.